### PR TITLE
Regulation Service

### PR DESCRIPTION
**Pull request by aushakov**

### CODE REVIEW

File: src/Lykke.Service.Regulation.AzureRepositories/ClientRegulationRepository.cs [hunk 1]
```diff
@@ -14,27 +18,126 @@ public ClientRegulationRepository(INoSQLTableStorage<ClientRegulationEntity> tab
             _tableStorage = tableStorage;
         }
 
-        public async Task<IClientRegulation> GetAsync(string clientId)
+        public async Task<IClientRegulation> GetAsync(string clientId, string regulationId)
         {
             var partitionKey = ClientRegulationEntity.GeneratePartitionKey();
-            var rowKey = ClientRegulationEntity.GenerateRowKey(clientId);
+            var rowKey = ClientRegulationEntity.GenerateRowKey(clientId, regulationId);
```
Comment (samodovdi) [inline src/Lykke.Service.Regulation.AzureRepositories/ClientRegulationRepository.cs]: the first parameter of GenerateRowKey, is regulationId and the second is clientId, but called vise versa

File: src/Lykke.Service.Regulation.AzureRepositories/ClientRegulationRepository.cs [hunk 2]
```diff
@@ -14,27 +18,126 @@ public ClientRegulationRepository(INoSQLTableStorage<ClientRegulationEntity> tab
...
+        {
+            IEnumerable<ClientRegulationEntity> entities = clientRegulations.Select(ClientRegulationEntity.Create);
+
+            return _tableStorage.InsertOrReplaceBatchAsync(entities);
+        }
+
+        public Task UpdateAsync(IClientRegulation clientRegulation)
+        {
+            var partitionKey = ClientRegulationEntity.GeneratePartitionKey();
+            var rowKey = ClientRegulationEntity.GenerateRowKey(clientRegulation.ClientId, clientRegulation.RegulationId);
```
Comment (samodovdi) [inline src/Lykke.Service.Regulation.AzureRepositories/ClientRegulationRepository.cs]: the same here

File: src/Lykke.Service.Regulation.AzureRepositories/ClientRegulationEntity.cs [hunk 3]
```diff
@@ -5,27 +5,36 @@ namespace Lykke.Service.Regulation.AzureRepositories
...
+
+        public bool Active { get; set; }
+
         internal static string GeneratePartitionKey()
         {
             return "ClientRegulation";
         }
 
-        internal static string GenerateRowKey(string clientId)
+        internal static string GenerateRowKey(string regulationId, string clientId)
```
Comment (samodovdi) [inline src/Lykke.Service.Regulation.AzureRepositories/ClientRegulationEntity.cs]: seems to be: string clientId, string regulationId

File: src/Lykke.Service.Regulation.AzureRepositories/Extensions/StorageExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+using AzureStorage;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Table;
+using Microsoft.WindowsAzure.Storage.Table.Protocol;
+
+namespace Lykke.Service.Regulation.AzureRepositories.Extensions
+{
+    public static class StorageExtensions
+    {
+        public static async Task InsertNoConflict<T>(this INoSQLTableStorage<T> storage, T entity) where T : ITableEntity, new()
```
Comment (samodovdi) [inline src/Lykke.Service.Regulation.AzureRepositories/Extensions/StorageExtensions.cs]: there is a method CreateIfNotExistsAsync on AzureTableStorage, could be used instead

File: src/Lykke.Service.Regulation/Startup.cs [hunk 5]
```diff
@@ -1,5 +1,4 @@
 ﻿using System;
```
Comment (samodovdi) [inline src/Lykke.Service.Regulation/Startup.cs:1]: Call Mapper.Configuration.AssertConfigurationIsValid(); after Mapper initialization

Comment (NovichikhinAlexey) [reply src/Lykke.Service.Regulation/Startup.cs:1]: fixed

**Pull request merged**